### DISCUSSION TITLE
Using .or() for columns of joined table

### DISCUSSION THREAD

**Discussion by dule-kitemetric** [supabase/supabase#28026] [Questions] [answered]
I have a problem when using `.or()` to filter multiple conditions on joined tables.

I found that the .or() only accept column names from the start table (in the below example is `groups`)

Example:
```
SELECT * FROM groups g
INNER JOIN group_members gm ON g.id = gm.group_id
WHERE
(gm.user_id = 1 AND gm.role = 'owner')
OR (gm.user_id = 2 AND gm.role = 'member');
```
(And there are maybe more conditions like that in the `WHERE` part)

How to do this using Supabase .or()? Or is there any solution?

I would be appreciated if someone can help me. Thank you so much.

**Comment by GaryAustin1** [comment, marked as answer]
In general the .or is all one table.  The main table or a single foreign table as an option.  You can't or across multiple tables.
https://github.com/orgs/supabase/discussions/18997 shows a potential workaround for some cases.

Likely you will need to do a view or rpc call to a postgres function.
